lib/piwik/test.js

```diff
@@ -119,6 +120,138 @@ describe('Piwik', function(){
+        analytics.didNotCall(window._paq.push, ['setCustomVariable', '1', 'UserId', '', 'page']);
+      });
+
+      it('shouldn\'t send more than the max number of custom variables', function(){
+        analytics.track('event',
+          { prop: true },
+          { integrations: {
+              Piwik: {
+                customVars: {
+                  1: ['UserId','6116'],
+                  2: ['SubscriptionId', ''],
+                  3: ['PlanName', 'ENTERPRISE'],
+                  4: ['New', 'item'],
+                  5: ['LastVariable', '0824'],
+                  6: ['TooManyVars', 'OhNo'],
+                  7: ['ThisIsntAThing', 'Apples']
+                }
+              }
+            }
+          }
+        );
+        analytics.called(window._paq.push, ['setCustomVariable', '1', 'UserId', '6116', 'page']);
+        analytics.called(window._paq.push, ['setCustomVariable', '2', 'SubscriptionId', '', 'page']);
+        analytics.called(window._paq.push, ['setCustomVariable', '3', 'PlanName', 'ENTERPRISE', 'page']);
+        analytics.called(window._paq.push, ['setCustomVariable', '4', 'New', 'item', 'page']);
+        analytics.called(window._paq.push, ['setCustomVariable', '5', 'LastVariable', '0824', 'page']);
+        analytics.didNotCall(window._paq.push, ['setCustomVariable', '6', 'TooManyVars', 'OhNo', 'page']);
+        analytics.didNotCall(window._paq.push, ['setCustomVariable', '7', 'ThisIsntAThing', 'Apples', 'page']);
+      });
+
+      it('should pick up the max variable setting change and send more than 5 variables', function(){
+        piwik.options.customVariableLimit = 10;
+        analytics.track('event',
+          { prop: true },
+          { integrations: {
+              Piwik: {
+                customVars: {
+                  1: ['UserId','6116'],
+                  2: ['SubscriptionId', ''],
+                  3: ['PlanName', 'ENTERPRISE'],
+                  4: ['New', 'item'],
+                  5: ['LastVariable', '0824'],
+                  6: ['TooManyVars', 'OhNo'],
+                  7: ['ThisIsntAThing', 'Apples']
+                }
+              }
+            }
+          }
+        );
+        analytics.called(window._paq.push, ['setCustomVariable', '1', 'UserId', '6116', 'page']);
+        analytics.called(window._paq.push, ['setCustomVariable', '2', 'SubscriptionId', '', 'page']);
+        analytics.called(window._paq.push, ['setCustomVariable', '3', 'PlanName', 'ENTERPRISE', 'page']);
+        analytics.called(window._paq.push, ['setCustomVariable', '4', 'New', 'item', 'page']);
+        analytics.called(window._paq.push, ['setCustomVariable', '5', 'LastVariable', '0824', 'page']);
+        analytics.called(window._paq.push, ['setCustomVariable', '6', 'TooManyVars', 'OhNo', 'page']);
+        analytics.called(window._paq.push, ['setCustomVariable', '7', 'ThisIsntAThing', 'Apples', 'page']);
+      });
     });
   });
 });
```

Comment:
Damn, I must have accidentally turned off that sublime extension you showed me for that